backends/platform/iphone/iphone_video.mm

```diff
@@ -657,62 +688,58 @@ - (bool)getMouseCoords:(CGPoint)point eventX:(int *)x eventY:(int *)y {
 	// thus we need to add it here.
 	*y = (int)(point.y * height + offsetY);
 
-	// Clip coordinates
-	if (*x < 0 || *x > width || *y < 0 || *y > height)
-		return false;
+	if (!iphone_touchpadModeEnabled()) {
+		// Clip coordinates
+		if (*x < 0 || *x > width || *y < 0 || *y > height)
+			return false;
+	}
 
 	return true;
 }
 
 - (void)deviceOrientationChanged:(UIDeviceOrientation)orientation {
-	switch (orientation) {
-	case UIDeviceOrientationLandscapeLeft:
-	case UIDeviceOrientationLandscapeRight:
-	case UIDeviceOrientationPortrait:
-		_orientation = orientation;
-		break;
+	[self addEvent:InternalEvent(kInputOrientationChanged, orientation, 0)];
+}
 
-	default:
-		return;
+- (UITouch *)secondTouchOtherTouchThan:(UITouch *)touch in:(NSSet *)set {
+	NSArray *all = [set allObjects];
+	for (UITouch *t in all) {
+		if (t != touch) {
+			return t;
+		}
 	}
-
-	[self addEvent:InternalEvent(kInputOrientationChanged, orientation, 0)];
+	return nil;
 }
 
 - (void)touchesBegan:(NSSet *)touches withEvent:(UIEvent *)event {
-	NSSet *allTouches = [event allTouches];
 	int x, y;
 
-	switch ([allTouches count]) {
-	case 1: {
-		UITouch *touch = [touches anyObject];
-		CGPoint point = [touch locationInView:self];
+	NSSet<UITouch *> *allTouches = [event allTouches];
```

Comment:
the « <UITouch *> » can be removed, this is only a new syntax element, introduced last year in clang, in order to provide information about the objects in the collection.